kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemMetadataProvider.java

```diff
@@ -534,6 +535,16 @@ static void writeDescriptor(
         throw new DatasetIOException("Cannot close", e);
       }
     }
+
+    //rename the file into its final location so that changes appear atomic
+      try {
```

Comment:
Nit: the formatting in this section is off. Formatting should use 2 spaces for an indent. (I know this is tedious, but it really helps with maintenance later.)